notifications/src/main/java/com/dojo/notifications/contest/Contest.java

```diff
@@ -13,25 +15,25 @@
     private String title;
     private String slackToken;
     private String slackChannel;
-    private List<String> senseiEmails;
+    private Set<String> senseiEmails;
 
     private Map<NotifierType, CommonNotificationsLevel> commonNotificationsLevel;
     private Set<NotifierType> personalNotifiers;
 
     public Contest() {
-        senseiEmails = new LinkedList<>();
+        senseiEmails = new HashSet<>();
         commonNotificationsLevel = new HashMap<>();
         commonNotificationsLevel.put(NotifierType.SLACK, CommonNotificationsLevel.NO_NOTIFICATIONS);
         commonNotificationsLevel.put(NotifierType.EMAIL, CommonNotificationsLevel.NO_NOTIFICATIONS);
         personalNotifiers = new HashSet<>();
     }
 
     public void setSenseiEmailsAsString(String emails) {
-        senseiEmails = Arrays.asList(emails.split(";"));
+        senseiEmails = Stream.of(emails.trim().split(";")).collect(Collectors.toSet());
```

Comment:
Using streams is more complex than just using Arrays.asList and wrap it in a Set collection